From 12f5ebdd8745beb2953f96a3ac4caf9a619ec3a3 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Wed, 6 Jul 2016 21:05:31 -0500
Subject: [PATCH 04/21] Accessing CRTC after miscellaneous output register
 initialization

When screen mode setting is being done for IGA1, VGA's miscellaneous
output register should be initialized before CRTC registers are
initialized.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index 0dcb659..f9341c1 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -885,9 +885,6 @@ viaIGA1SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
     xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                 "IGA1 Requested Screen Mode: %s\n", mode->name);
 
-    ViaCrtcMask(hwp, 0x11, 0x00, 0x80); /* modify starting address */
-    ViaCrtcMask(hwp, 0x03, 0x80, 0x80); /* enable vertical retrace access */
-
     /* Set Misc Register */
     temp = 0x23;
     if (mode->Flags & V_NHSYNC)
@@ -911,11 +908,13 @@ viaIGA1SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
     hwp->writeSeq(hwp, 0x03, 0x00);
     hwp->writeSeq(hwp, 0x04, 0x0E);
 
+    ViaCrtcMask(hwp, 0x03, 0x80, 0x80); /* enable vertical retrace access */
 
     /* Setting maximum scan line to 0. */
     /* 3X5.09[4:0] - Maximum Scan Line */
     ViaCrtcMask(hwp, 0x09, 0x00, 0x1F);
 
+    ViaCrtcMask(hwp, 0x11, 0x00, 0x80); /* modify starting address */
 
     /* 3X5.14[6]   - Double Word Mode
      *               Allows normal addressing or double-word addressing.
-- 
1.9.1

